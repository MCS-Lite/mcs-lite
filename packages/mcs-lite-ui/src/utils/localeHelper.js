// @flow
import R from 'ramda';

type LocaleMapper = string => string => string;
type GetMCSLinkByLocale = string => string;

export const LOCALES = [
  {
    id: 'en',
    children: 'English',
  },
  {
    id: 'zh-TW',
    children: '繁體中文',
  },
  {
    id: 'zh-CN',
    children: '简体中文',
  },
];

export const localeMapper: LocaleMapper = defaultLocale =>
  R.memoize(string =>
    R.cond([
      [R.test(/^en/gi), R.always('en')],
      [R.test(/^zh-TW/gi), R.always('zh-TW')],
      [R.test(/^zh-CN/gi), R.always('zh-CN')],
      [R.test(/^zh/gi), R.always('zh-TW')],
      [R.T, R.always(defaultLocale)],
    ])(string),
  );

export const getMCSLinkByLocale: GetMCSLinkByLocale = R.memoize(locale =>
  R.cond([
    [R.equals('en'), R.always('https://mcs.mediatek.com/')],
    [R.equals('zh-CN'), R.always('https://mcs.mediatek.cn/')],
    [R.T, l => `https://mcs.mediatek.com/${l}/`],
  ])(locale),
);
