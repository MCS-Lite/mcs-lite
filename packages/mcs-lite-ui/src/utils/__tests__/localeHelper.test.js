// @flow

import { LOCALES, localeMapper, getMCSLinkByLocale } from '../localeHelper';

it('should return LOCALES', () => {
  expect(LOCALES).toMatchSnapshot();
});

it('should return correct locale with localeMapper', () => {
  const defaultLocaleMapper = localeMapper('zh-TW');

  expect(defaultLocaleMapper('zh-tW')).toBe('zh-TW');
  expect(defaultLocaleMapper('zh-tw')).toBe('zh-TW');
  expect(defaultLocaleMapper('zh-TW')).toBe('zh-TW');
  expect(defaultLocaleMapper('zh-CN')).toBe('zh-CN');
  expect(defaultLocaleMapper('zh-cN')).toBe('zh-CN');
  expect(defaultLocaleMapper('zh-cn')).toBe('zh-CN');
  expect(defaultLocaleMapper('zh')).toBe('zh-TW');
  expect(defaultLocaleMapper('zH')).toBe('zh-TW');
  expect(defaultLocaleMapper('ZH')).toBe('zh-TW');
  expect(defaultLocaleMapper('en')).toBe('en');
  expect(defaultLocaleMapper('en-US')).toBe('en');
  expect(defaultLocaleMapper('EN')).toBe('en');
  expect(defaultLocaleMapper('ja')).toBe('zh-TW');
  expect(defaultLocaleMapper('others')).toBe('zh-TW');
});

it('should return correct links with getMCSLinkByLocale', () => {
  expect(getMCSLinkByLocale('zh-TW')).toBe('https://mcs.mediatek.com/zh-TW/');
  expect(getMCSLinkByLocale('zh-CN')).toBe('https://mcs.mediatek.cn/');
  expect(getMCSLinkByLocale('en')).toBe('https://mcs.mediatek.com/');
});
