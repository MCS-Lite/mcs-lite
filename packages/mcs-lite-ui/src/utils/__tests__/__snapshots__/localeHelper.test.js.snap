// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return LOCALES 1`] = `
Array [
  Object {
    "children": "English",
    "id": "en",
  },
  Object {
    "children": "繁體中文",
    "id": "zh-TW",
  },
  Object {
    "children": "简体中文",
    "id": "zh-CN",
  },
]
`;
